### ISSUE TITLE
USWDS - Bug: File input issues with voice command and screen readers

### ISSUE THREAD

**Issue by amyleadem** [uswds/uswds#5192]
## Describe the bug

Testing revealed issues when using the file input component with various assistive technologies.

Thanks to @briandeconinck for flagging this in https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1499#issuecomment-1458661931.

More details about the issues can be found in this [Google doc](https://docs.google.com/document/d/1kqN166hHXDSkmsdHXL5dtD8u0nkHD5oJzVuptiY0Yhw/edit#) :lock:

### Screen readers:
The input `aria-label` provides insufficient information for interaction. This default `aria-label` does not clearly communicate what the input is or what the user should do, which can lead to confusion. 

###  Voice command: 
Voice command does not work in an intuitive way. A user would expect to say something like "Click drag file here or choose from folder" or "Click choose from folder" to interact with the file input because that is the text that is visible in the component. 

Instead, the only ways to interact with the component through voice are to either:
- Give a command using the text found in the `aria-label`  
- Work with the mouse grid or other cumbersome mouse-based voice commands.

Both of these solutions would be very frustrating for a VC user. 

## Steps to reproduce the bug

Using Voice command software Dragon NaturallySpeaking, attempt to open the file input component with voice command. 

Intuitive methods:
- Reading the visible component text - does not work:
    - “Click choose from folder” is not recognized.
    - "Click drag file here or choose from folder" is not recognized.

Alternative methods:
- Reading the aria-label - works: 
    - Voice command “Click no file selected” opens the file selection window.
    - Voice command “Click cancel”; closes the file selection window.
    - However, the `aria-label` is not visible and a user would not intuitively find this.
- Identifying all links - does not work:
    - “Click link” will highlight all links and assign each a number, but this does not recognize file input 
- Using mouse grid or other mouse commands - works, but is a frustrating experience:
    - “Open mouse grid” should open a grid but during the demo it proved to be frustrating. 
    - “Move mouse up” will move the mouse until you tell it to stop; this is a method of last resort. 

> Demonstration with Dragon:

 > https://user-images.githubusercontent.com/6844775/223518739-84b4b68e-3312-4669-a8b5-d58a3f0f60e6.mp4

> _Originally posted by @briandeconinck in https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1499#issuecomment-1458661931_

## Expected Behavior

1. Voice control users should be able to give commands using the visible text to move focus to and interact with the input.
    - The voice command “Click, [visible text]” should move focus into an input and allow the user to add content. 
        - For example, in the current file input example, users would expect to say "Click drag file here or choose from folder" or “Click choose from folder” to interact with the component.
1. Screen reader users should be given clear and explicit labels and instructions. 
    - Labels should give information about the role of a given input and should give users instructions about what action they must take to interact with the input.

## Additional context
- We should confirm that `aria-label` takes precedence over `label in our available screen readers.

**Comment by amyleadem** [comment]
Note: Tested with VoiceOver in Safari and Chrome and found that it reads the label, hint, and aria-label. Noting specifically that the aria-label does not override the label.

**Comment by amyleadem** [comment]
@amycole501 would you be able to test the [file input component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-form-inputs-file-input--multiple&viewMode=story) in NVDA and JAWS and report back if it reads the label, hint, and/or aria-label?

**Comment by amycole501** [comment]
Add a period after each sentence since JAWS tries to smush both sentences together.
Input accepts multiple files.
Select one or more files.

In JAWS I hear "no file selected button" when the file box area is selected.
 I don't hear "Drag files here or choose from folder" nor do I hear that there's a link. 

I think it's OK as long  as the person can tell "This is where you upload a file. Choose a file here."

In NVDA I hear "Button no file selected" when the file box area is selected.

What might help is adding "Select one or more files using the upload feature below." It will work to tell visually impaired users that coming up next is the file upload feature and will clarify that "select one or more files" is NOT the action, the action takes place below.
